src/ja/user/cli.py

```diff
@@ -1,19 +1,210 @@
+        parser_query.add_argument("--status",
+                                  nargs="+",
+                                  choices="new queued running paused done killed crashed 0 1 2 3 4 5 6".split(),
+                                  help="Job status(es) to filter query results by.")
+        parser_query.add_argument("--preemptible", "--is-preemptible", "--ip",
+                                  dest="preemptible",
+                                  choices="false true t f".split(),
+                                  help="Job.is_preemptible value to filter query results by.")
+        parser_query.add_argument("--special-resources", "--sr", nargs="+")
+        parser_query.add_argument("--threads", "-t",
+                                  nargs=2,
+                                  type=int,
+                                  help="Lower and upper bound of job CPU thread count to filter query results by.")
+        parser_query.add_argument("--memory", "-m",
+                                  nargs=2,
+                                  type=int,
+                                  help="Lower and upper bound of job memory allocation in MB\
+                                        to filter query results by.")
+        parser_query.add_argument("--before",
+                                  help="Jobs scheduled before this point in time (YYYY-MM-DD[ HH:MM:SS]).")
+        parser_query.add_argument("--after",
+                                  help="Jobs scheduled after this point in time (YYYY-MM-DD[ HH:MM:SS]).")
+
+        # parser for cancel command
+        parser_cancel = subparsers.add_parser("cancel")
+        parser_cancel.add_argument("label_or_uid",
+                                   choices=["label", "uid"],
+                                   help="You can cancel a job by specifying its uid OR its label")
+        parser_cancel.add_argument("arg", help="The uid or label of the job that needs to be cancelled.")
+
+        args: Namespace = parser.parse_args(cli_args)
+        ssh_config = SSHConfig(args.hostname, args.username, args.password, args.key_path, args.passphrase)
+        user_config = UserConfig(ssh_config, Verbosity(args.verbosity))
+
+        if args.command == "add":
+            # creating job instance
+            constraints: JobSchedulingConstraints = JobSchedulingConstraints(UserClientCLIHandler._pr[args.priority],
+                                                                             args.preemptible,
+                                                                             args.special_resources)
+            docker_constraints: DockerConstraints = DockerConstraints(args.threads, args.memory)
+            mount_points: List[MountPoint] = []
+            for i in range(0, len(args.mount), 2):
+                mount_points.append(MountPoint(args.mount[i], args.mount[i + 1]))
+            docker_context: DockerContext = DockerContext(args.source, mount_points)
```

Comment:
You are giving the source file **path** to the DockerContext constructor.
However, the DockerContext constructor expects to be given the source file **contents**.
You need to read from the source file and pass the contents on to the DockerContext constructor.

Reply:
Now the whole file is read and passed on to the constructor as a string, hope that's right